src/transformers/trainer.py

```diff
@@ -552,32 +514,24 @@ def train(self, model_path: Optional[str] = None):
         logger.info("\n\nTraining completed. Do not forget to share your model on huggingface.co/models =)\n\n")
         return TrainOutput(self.global_step, tr_loss / self.global_step)
 
-    def _log(self, logs: Dict[str, float], iterator: Optional[tqdm] = None) -> None:
-        if self.epoch is not None:
-            logs["epoch"] = self.epoch
-        if self.tb_writer:
-            for k, v in logs.items():
-                if isinstance(v, (int, float)):
-                    self.tb_writer.add_scalar(k, v, self.global_step)
-                else:
-                    logger.warning(
-                        "Trainer is attempting to log a value of "
-                        '"%s" of type %s for key "%s" as a scalar. '
-                        "This invocation of Tensorboard's writer.add_scalar() "
-                        "is incorrect so we dropped this attribute.",
-                        v,
-                        type(v),
-                        k,
-                    )
-            self.tb_writer.flush()
-        if is_wandb_available():
-            if self.is_world_master():
-                wandb.log(logs, step=self.global_step)
-        output = json.dumps({**logs, **{"step": self.global_step}})
-        if iterator is not None:
-            iterator.write(output)
-        else:
-            print(output)
+    def _log_tb(self, logs: Dict[str, float]) -> None:
+        """
+        Log data to tensorboard. Called by `log_metrics`.
+        """
+        for k, v in logs.items():
+            if isinstance(v, (int, float)):
+                self.tb_writer.add_scalar(k, v, self.global_step)
+            else:
```

Comment:
In case the value is a string.

```suggestion
            elif isinstance(v, str):
                tf.summary.text(k, v, self.global_step)
            else:
```
Or even better you can cast the value to `str` to log it and log the error in case it cannot be cast.